### DISCUSSION TITLE
Pull Shark: not all PR's listed

### DISCUSSION THREAD

**Discussion by ExposedCat** [community/community#18298] [Other Feature Feedback, Questions, & Ideas]
Just found the Beta version of `Pull Shark` achievement in my profile and some of my merged pull requests are not listed there. So is it a bug or there are some requirements for requests to be shown there?

**Comment by schneedotdev** [comment]
The Achievement doesn't level up for every single pull request merged, there are certain numbers you need to hit in order to rank up. A list of those can be found here: https://github.com/Schweinepriester/github-profile-achievements

**Reply by schneedotdev** [reply]
For example:

Pull Shark Bronze: 16 pull requests merged
Pull Shark Silver: 128 pull requests merged
Pull Shark Gold: 1024 pull requests merged

**Reply by ExposedCat** [reply]
Hm, yes, but all of the requests display as a chain there, does not it? I have much more than 2 requests shown there. Moreover, there are only my requests to my repos, and no any to other repos LOL :)

**Reply by curtinaz** [reply]
I think 1024 PR's seems a little bit astronomical

**Reply by Rohlik** [reply]
@curtinaz Yes and no :) For example, when you are working for some company which use GitHub for everything, then it is not that hard to get Gold tier of this achievement. Of course, maybe it will takes 2-3 year but still this tier shouldn't be granted to everyone.

**Reply by salman-ryo** [reply]
I'm not sure if it is a bug but only 1 PR is counted per repository

**Reply by tetuaoro** [reply]
> I'm not sure if it is a bug but only 1 PR is counted per repository

No, it counts all the merged PR.

**Comment by aexvir** [comment]
at least for me there does seem to be some kind of bug on how it's counted, as I have hundreds of merge requests between 2016 and 2018, when I used to work at a company using github, that are not counted; github shows as my second pr some random change done in jan 2018

the only things that I can think could be messing with this are the fact that I changed my username back then, and that the organization I used to work on might not be there anymore (I left the org when I left the company, so can't tell for sure)

I mean... no big deal I guess... but it's a shame that this number is not accurate; I'd have expected that GitHub should have a proper registry of how many and what merge requests have been done by a user, regardless of the change of username or the visibility of the organization

**Reply by Putt0** [reply]
Look this: https://github.com/Schweinepriester/github-profile-achievements

**Comment by CamiloCortesM** [comment]
how to unlock this achievement?

**Reply by Madushan98** [reply]
You got to open pull requests that have been merged

**Reply by CamiloCortesM** [reply]
how to unlock the yolo achievement?

**Reply by Madushan98** [reply]
yolo is archived when you merge a r without reviewing it

**Reply by CamiloCortesM** [reply]
and how can you do that at this moment I am studying github well but I am just starting out

**Reply by ottogugel** [reply]
You must open an order that has been merged

**Reply by dedysteady** [reply]
thanks

**Comment by jisaacks** [comment]
Definitely not counting mine either. Says I earned my 2nd bronze badge in 2014 and nothing since. I have definitely had 100s of PRs merged in the past 8 years of daily use.

**Reply by markstos** [reply]
Maybe the Pull Shark award means PRs merged in repos you don't control, or that are no part of your organization-- in other words, primary open source contributions?

**Reply by jisaacks** [reply]
@markstos I doubt it. I see colleagues getting credit for them. I have made several open source PRs as well. But would love a definitive answer and some transparency.

**Comment by vhirtham** [comment]
I think what causes the most confusion (at least for me) are the PRs that are shown when you hover over the achievement. These are not all the PRs that are counted but the ones that got you to the next level. Additional confusion comes from the fact that bronze, silver, and gold badge are not constantly displayed as depicted [here](https://github.com/Schweinepriester/github-profile-achievements#achievements). You only see them for a short time in the animation when hovering over the symbol.

So in my case, I see the standard blue symbol with an `x3` attached. When I move the mouse over the symbol I see 3 PRs mentioned and an animation I didn't pay much attention to. So I initially thought there were only 3 PRs counted. However, if you take a closer look, all those 3 PRs have a small note behind them `2nd pull request merged`, `16th pull request merged`, `128th pull request merged` etc. Those are the numbers that get you to the next level. The mentioned PR is the one that let you progress to the next level.
The animation in my profile first shows the silver badge, then the bronze badge, and finally the default symbol. So the counting seems to be correct (in my case), it is just the presentation of the badge that is somewhat unintuitive.

When I visit the profile of @ExposedCat , I come to the conclusion that he was also confused by this. If you scroll down in the profile he has a small GitHub stat table that says he has merged 30 PRs. The Achievement reflects this by awarding him the bronze version (2x at the achievement symbol, the 2 "milestone PRs" listed, and a bronze badge at the start of the animation). So everything seems to be correct there.

**Suggestion**

I think all this confusion can be avoided if the badge is redesigned to be more intuitive:

- the static symbol should directly reflect the current status. Show the bronze, silver, or gold image in the profile once achieved and not the default symbol with a multiplier attached to it
- Flip the order of the animation. IMHO it doesn't make sense to show the highest achievements just for a second and then the lowest one for the rest of the time. It should start at the lowest and raise to the current level which is then displayed continuously. This alone would probably help a lot with the confusion.
- Add a counter of the total PRs and maybe a target number for the next level. Place it over the "history" so that it is immediately visible how many PRs were counted. 
- Maybe make the "history" erties a bit more descriptive: "Achived bronze status by merging his 16th pull request: LINK_TO_PR"

Here is a [**related suggestion in a separate topic**](https://github.com/github-community/community/discussions/19473)

**Reply by jisaacks** [reply]
It may be correct for you, but it’s not for me. I merge PRs daily, I’ve been a member for over a decade. I earned my last badge 8 years ago and nothing since? You do the math, it doesn’t add up.

**Reply by vhirtham** [reply]
Didn't say that it is true in all cases. But at least for @ExposedCat, who opened this issue, it seems that he just got confused by the badge's design. 

In your case, I can't really tell what the problem is. GitHub has definitely some issues/bugs with its contribution counting. I have a public organization repo where this is pretty obvious.

It might help if you add [this stats table](https://github.com/anuraghazra/github-readme-stats) to your profile, at least temporarily. I think it uses the same counts and you can see how many PRs have been counted so far. Then just check if the number increases next time you merge a PR.
